edb/pgsql/resolver/context.py

```diff
@@ -72,6 +72,13 @@ class Scope:
     # Common Table Expressions
     ctes: List[CTE] = field(default_factory=lambda: [])
 
+    # Pairs of columns of the same name that have been compared in a USING
+    # clause. This makes unqualified references to their name them un-ambagious,
+    # it resolves to the column in the left (first) table.
```

Comment:
Does it do the left table for RIGHT JOIN and FULL OUTER JOIN as well?

Reply:
No, it looks like for RIGHT JOIN it uses the right key and for FULL OUTER JOIN it uses whichever of them is non-NULL.
So I guess we need to be sensitive to what the join type is, and take the right for RIGHT JOIN and take `coalesce(left, right)` for FULL OUTER JOIN?
(Or maybe just decide we don't care much and always do the coalesce??)

Reply:
coalescing might interfere with index use, so need to be cognizant of that